File: bug66_empty_case_failure/bug66_empty_case_failure/main1.js
function main() {
    let data = [];
    let total = 0;

    for (let v of data) {
        total += v;
    }

    let avg;

    if (data.length === 0) {
        avg = 0; // ✅ FIX
    } else {
        avg = total / data.length;
    }

    let acc = (BigInt(Math.trunc(avg)) * 0x9E3779B97F4A7C15n) & 0xFFFFFFFFFFFFFFFFn;

    console.log(acc.toString(16)); // ✅ Added output
}
main();

File: bug66_empty_case_failure/bug66_empty_case_failure/main1.java
public class main1 {
    public static void main(String[] args) {
        int[] data = new int[0];
        int total = 0;

        for (int v : data) {
            total += v;
        }

        double avg;

        if (data.length == 0) {
            avg = 0; // ✅ FIX
        } else {
            avg = (double) total / data.length;
        }

        long acc = ((long) avg * 0x9E3779B97F4A7C15L);

        System.out.println(Long.toHexString(acc)); // ✅ Added output
    }
}